pkg/glanceapi/statefulset.go

```diff
@@ -290,19 +296,32 @@ func StatefulSet(
 		extraVolPropagation),
 		apiVolumes...)
 
-	// If possible two pods of the same service should not
-	// run on the same worker node. If this is not possible
-	// the get still created on the same worker node.
-	statefulset.Spec.Template.Spec.Affinity = affinity.DistributePods(
-		common.AppSelector,
-		[]string{
-			glance.ServiceName,
-		},
-		corev1.LabelHostname,
-	)
 	if instance.Spec.NodeSelector != nil {
 		statefulset.Spec.Template.Spec.NodeSelector = *instance.Spec.NodeSelector
 	}
 
+	if topology != nil {
+		ts := topology.Spec
+		if ts.TopologySpreadConstraints == nil {
+			ts.TopologySpreadConstraints = &tplg
```

Comment:
I don't understand what this is doing.
-  Is the reference to "&tplg" just to an empty array per L80?
-  How does this affect the statefulset?

Reply:
Long time I don't check this part. I think the main problem I had was to make sure we do not dereference something that is not present in the Topology CR to avoid segfaults. This was the high level idea, but, considering we don't have an Affinity override that is applied regardless of the TopologySpreadConstraint, I think I can really simplify this part further.
Let me push an update to make sure this is simple enough and easy to read.

Reply:
Done, my last update simplifies this part and improves readability.